File: pyg_parent1/pyg_sellergoods_interface/src/main/java/com/pyg/sellergoods/service/BrandService.java
package com.pyg.sellergoods.service;

import com.pyg.pojo.TbBrand;
import entity.PageResult;

import java.util.List;
import java.util.Map;

public interface BrandService {
    //需求：查找所有品牌信息
    public List<TbBrand> findAll();
    //需求：分页查找
    public PageResult findPage(int pageNum,int pageSize);
    //需求：添加
    public void add(TbBrand tbBrand);
    //单个查询品牌
    public TbBrand findOne(Long id);
    //需求：修改
    public void update(TbBrand tbBrand);
    //需求：删除
    public void delete(Long[] ids);
    //需求：按条件查询
    public PageResult findPage(TbBrand brand,int pageNum,int pageSize);
    public List<Map> selectOptionList();
}

File: pyg_parent1/pyg_manage_web/src/main/webapp/js/service/brandService.js
//自定义服务
app.service('brandService',function ($http) {
    this.findAll=function () {
        return $http.get('../brand/findAll.do')
    }
    //分页
    this.findPage=function (page,rows) {
        return $http.get('../brand/findPage.do?page='+page+'&rows='+rows);
    }
    this.findOne=function (id) {
        return  $http.get('../brand/findOne.do?id='+id);
    }
    this.dele=function (ids) {
        return $http.get('../brand/delete.do?ids='+ids);
    }
    this.search=function (page,rows,entity) {
        return $http.post('../brand/search.do?page='+page+'&rows='+rows, entity);
    }
    this.add=function (entity) {
        return  $http.post('../brand/add.do',entity);
    }
    this.update=function (entity) {
        return  $http.post('../brand/update.do',entity);
    }
    this.selectOptionList=function () {
        return $http.get('../brand/selectOptionList.do');
    }
})